keps/sig-storage/5381-mutable-pv-affinity/README.md

```diff
@@ -0,0 +1,872 @@
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to make `PersistentVolume.spec.nodeAffinity` field mutable,
+making it possible to change the affinity with VolumeAttributeClass.
+This allows user to migrate data or enabling features without
+interrupting workloads.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Currently, `PersistentVolume.spec.nodeAffinity` is set at creation time and cannot be changed.
+But user may modify the volume to
+taking advantage of new features provided by the storage provider,
+or accommodate to the changes of business requirements.
+These modification can be expressed by `VolumeAttributeClass` in Kubernetes.
+But sometimes, A modification to volume comes with change to its accessibility, such as:
+1. migration of data from one zone to regional storage;
+2. enabling features that is not supported by all the client nodes.
```

Comment:
Though it deviates from nodeAffinity's original design（geography related）, this scenario is currently useful for users, hence we include it.

Reply:
How do we prevent the new update to the nodeAffinity is not compatible with the current node?

Reply:
Or maybe we have some rules around you can only add wider nodeAffinity support but not shrink it.